bin/oref0-pump-loop.sh

```diff
@@ -3,76 +3,73 @@
 # main pump-loop
 main() {
     prep
-    until( \
-        echo && echo Starting pump-loop at $(date): \
-        && wait_for_bg \
-        && wait_for_silence \
-        && if_mdt_get_bg \
-        && refresh_old_pumphistory_enact \
-        && refresh_old_pumphistory_24h \
-        && refresh_old_profile \
-        && touch monitor/pump_loop_enacted -r monitor/glucose.json \
-        && refresh_temp_and_enact \
-        && refresh_pumphistory_and_enact \
-        && refresh_profile \
-        && refresh_pumphistory_24h \
-        && echo Completed pump-loop at $(date) \
+    echo && echo "Starting pump-loop at $(date):" \
+    && wait_for_bg \
+    && wait_for_silence \
+    && if_mdt_get_bg \
+    && refresh_old_pumphistory_enact \
+    && refresh_old_pumphistory_24h \
+    && refresh_old_profile \
+    && touch monitor/pump_loop_enacted -r monitor/glucose.json \
+    && refresh_temp_and_enact \
+    && refresh_pumphistory_and_enact \
+    && refresh_profile \
+    && refresh_pumphistory_24h 
+    if [ $? -eq 0 ]; then
+        echo "Completed pump-loop succesfully at $(date)" \
         && touch monitor/pump_loop_completed -r monitor/pump_loop_enacted \
-        && echo); do
-
-            # On a random subset of failures, wait 45s and mmtune
-            echo Error, retrying \
-            && maybe_mmtune
-            sleep 5
-    done
+        && echo
+    else
+        echo "Aborted pump-loop with exitcode $? at $(date)" \
+        echo Waiting 45s and maybe mmtune \
+        sleep 45 && maybe_mmtune && sleep 5
+    fi
 }
 
 # main supermicrobolus loop
 smb_main() {
     prep
-    until ( \
```

Comment:
I've been running with this configuration for about a month now without problems. I can't see any reason why removing the until creates a problem. I can think of various reasons when the until creates problems. At least WW-pumps seem to be not happy of killing processes half way and I had situation were a battery reset and mmtune was needed to make sure pump communication was ok again.

All other loops run only once, and get initiated with a minute-based cron.